File: Codeforces/collider.java
import java.util.Scanner;

public class collider {
    public static boolean isRelaPrime(int a, int b){
        int hcf = 0;
        for(int i=1; i<=a|| i<=b ; i++){
            if(a%i == 0 && b%i == 0){
                hcf = i;
            }
        }
        if(hcf == 1)
        return true;
        else
        return false;
    }
        public static void main(String[] args) {
            Scanner sc = new Scanner(System.in);
            int n = sc.nextInt();
            int m = sc.nextInt();

            boolean[] isOn = new boolean[n+1];
            for(int i=1; i<=n ; i++){
            isOn[i] = false;
            }
            char sign1 = sc.next().charAt(0);
            int a1 = sc.nextInt();

            if(sign1 == '+'){
                isOn[a1] = true;
                System.out.println("Success");
            }else{
                System.out.println("Already off");
            }

            for(int i=0; i<m-1 ; i++){
                char sign = sc.next().charAt(0);
                int a = sc.nextInt();


            boolean check = true ; 
            for(int k=1; k<=n ; k++){
                check = isOn[k];

                if(check)
                break;
            }
            if(check){
            if(sign == '+'){

                if(isOn[a])
                System.out.println("Already on");

                for(int j=1; j<=n ; j++){
                    if(isOn[j]){
                        if(isRelaPrime(a, j) == false){
                            System.out.println("Conflict with "+j);
                        }else{
                            isOn[a] = true;
                            System.out.println("Success");
                        }
                    }
                }

            }if(sign == '-'){
                if(isOn[a] == false)
                System.out.println("Already off");
                else{
                    isOn[a] = false;
                System.out.println("Success");
            }
            }
        }else{
            isOn[a] = true;
                System.out.println("Success");
        }
            }
            
            sc.close(); 
            }
        }
        
        

    

File: Codeforces/collider.cpp
#include<bits/stdc++.h>
using namespace std;
#define MAX 100002

vector<int> prime(MAX, 1);
vector<int> primeDivisor[MAX];

void sieve() {
    for(int i = 2; i<= MAX; i++) {
        if(prime[i]) {
            for(int j = 2 * i; j <= MAX; j += i) {
                prime[j] = 0;
                primeDivisor[j].push_back(i);
            }
        }
    }
}

int main () {
    int n, m;
    cin >> n >> m;
    sieve();

    vector<int> collider(n+1, 0);
    
    while (m--) {
        char c;
        int k;
        cin >> c >> k;

        if(c == '+') {
            if(collider[k] == k) {
                cout << "Already on\n";
                continue;
            }

            if(collider[k]) {
                cout << "Conflict with " << collider[k] << "\n";
                continue;
            }

            for(int i = 0 ; i < primeDivisor[k].size(); i++) {
                int divisor = primeDivisor[k][i];
                if(collider[divisor]) {
                    cout << "Conflict with " << collider[divisor] << "\n";
                    goto end;
                }
            }

            for(int i = 0 ; i < primeDivisor[k].size(); i++) {
                int divisor = primeDivisor[k][i];
                collider[divisor] = k;
            }

            collider[k] = k;
            cout << "Success\n";
        } else {
            if(collider[k] != k) {
                cout << "Already off\n";
            } else {
                for(int i = 0 ; i < primeDivisor[k].size(); i++) {
                    int divisor = primeDivisor[k][i];
                    collider[k] = divisor;
                }
                collider[k] = 0;
                cout << "Success\n";
            }
        }
        end: ;
    }

}